htdocs/website/index.php

```diff
@@ -2155,6 +2155,65 @@
 	}
 }
 
+$form = new Form($db);
+$formadmin = new FormAdmin($db);
+$formwebsite = new FormWebsite($db);
+$formother = new FormOther($db);
+$domainname = '0.0.0.0:8080';
+$tempdir = $conf->website->dir_output.'/'.$websitekey.'/';
+
+// Confirm generation of website sitemaps
+if($action == 'confirmgeneratesitemaps'){
+	$formconfirm = $form->formconfirm($_SERVER["PHP_SELF"].'?website='.$website->ref, $langs->trans('ConfirmSitemapsCreation'), $langs->trans('ConfirmGenerateSitemaps', $object->ref),'generatesitemaps', '', "yes", 1);
+	$action = 'preview';
+}
+
+// Generate web site sitemaps
+if ($action == 'generatesitemaps') {
+	$domtree = new DOMDocument('1.0', 'UTF-8');
+	$root = $domtree->createElementNS('http://www.sitemaps.org/schemas/sitemap/0.9', 'urlset');
+	$domtree->formatOutput = true;
+
+	$sql = "SELECT wp.type_container , wp.pageurl, wp.lang, DATE(wp.tms) as tms, w.virtualhost";
+	$sql .= " FROM ".MAIN_DB_PREFIX."website_page as wp, ".MAIN_DB_PREFIX."website as w";
+	$sql .= " WHERE wp.type_container IN ('page', 'blogpost')";
+	$resql = $db->query($sql);
+	if ($resql) {
+		$num_rows = $db->num_rows($resql);
+		if ($num_rows > 0) {
+			$i = 0;
+			while ($i < $num_rows) {
+				$objp = $db->fetch_object($resql);
+				$url = $domtree->createElement('url');
+				$pageurl = $objp->pageurl;
+				if ($objp->lang) {
+					$pageurl = $objp->lang.'/'.$pageurl;
+				}
+				if ($objp->virtualhost) {
+					$domainname = $objp->virtualhost;
+				}
+				$loc = $domtree->createElement('loc', 'http://'.$domainname.'/'.$pageurl);
+				$lastmod = $domtree->createElement('lastmod', $objp->tms);
+
+				$url->appendChild($loc);
+				$url->appendChild($lastmod);
+				$root->appendChild($url);
+				$i++;
+			}
+			$domtree->appendChild($root);
+			if ($domtree->save($tempdir.'sitemaps.'.$websitekey.'.xml'))
+			{
+				setEventMessages($langs->trans("SitemapGenerated"), null, 'mesgs');
```

Comment:
After having generated the sitemap, it may be interesting to add this line into the robots.txt file 
"Sitemap: http://example.com/sitemap_location.xml"
So sitemap will be detected and took into account. You can for this:
- Read existing robots.txt file with
 		$robotcontent = @file_get_contents($filerobot);
		// Clean the php htmlheader file to remove php code and get only html part
		$robotcontent = preg_replace('/<\?php \/\/ BEGIN PHP[^\?]*END PHP \?>\n*/ims', '', $robotcontent);
- rewrite it after adding the line
"Sitemap: http://example.com/sitemap_location.xml"  (if a line Sitemap is already present, replace it)


call dolSaveRobotFile('robots.txt');